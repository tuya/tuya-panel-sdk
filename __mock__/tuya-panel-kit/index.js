import React from 'react';
import { View } from 'react-native';
import TYText from './components/TYText';
import UnitText from './components/UnitText';

module.exports = {
  TYText,
  UnitText,
  Utils: {
    RatioUtils: {
      convertX: () => {},
    },
  },
};
