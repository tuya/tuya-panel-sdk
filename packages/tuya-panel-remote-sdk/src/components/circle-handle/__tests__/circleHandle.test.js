import React from 'react';
import { shallow } from 'enzyme';
import { Utils, IconFont } from 'tuya-panel-kit';
import { Image, Text } from 'react-native';
import CircleHandle from '../index';
import icon from './res/iconfont.json';

const { convertX: cx } = Utils.RatioUtils;
const add = require('./res/add.png');
const loading = require('./res/loading.png');

function customizeLoading(props = {}) {
  const wrapper = shallow(<CircleHandle loadingContent={<Image source={loading} />} {...props} />);
  const instance = wrapper.instance();
  return { wrapper, instance };
}

function iconLoading(props = {}) {
  const wrapper = shallow(
    <CircleHandle
      keyContent={{
        top: <Text style={{ color: 'rgba(206,89,75,1)' }}>{'top'}</Text>,
        right: <IconFont d={icon.minus} size={cx(18)} />,
        bottom: <Text style={{ color: 'rgba(85,158,101,1)' }}>{'bottom'}</Text>,
        left: <Image source={add} style={{ tintColor: '#CCC' }} />,
      }}
      {...props}
    />
  );
  const instance = wrapper.instance();
  return { wrapper, instance };
}

describe('PressKey components', () => {
  it('content render', () => {
    const wrapper = shallow(
      <CircleHandle
        centerTextStyle={{ fontSize: cx(16), color: 'red' }}
        keyContent={{
          top: <Text style={{ color: 'rgba(206,89,75,1)' }}>{'top'}</Text>,
          right: <IconFont d={icon.minus} size={cx(18)} />,
          bottom: <Text style={{ color: 'rgba(85,158,101,1)' }}>{'bottom'}</Text>,
          left: <Image source={add} style={{ tintColor: '#CCC' }} />,
        }}
        tip={{
          top: 'top',
          right: 'right',
          bottom: 'bottom',
          left: 'left',
        }}
        tipStyle={{ color: 'rgba(194,117,241,1)' }}
        status={{
          top: true,
          bottom: true,
          left: true,
          right: true,
          center: true,
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('customize render', () => {
    const wrapper = shallow(
      <CircleHandle
        padding={cx(10)}
        outBgColor="rgba(246,206,95,0.3)"
        radius={cx(90)}
        centerRadius={cx(30)}
        offset={cx(20)}
        pointColor="#FFF"
        centerText="OK"
        pointRadius={cx(4)}
        centerTextStyle={{ fontSize: cx(16), color: 'white' }}
        bgColor={{
          top: 'rgba(206,89,75,1)',
          right: 'rgba(246,206,95,1)',
          bottom: 'rgba(85,158,101,1)',
          left: 'rgba(87,140,238,1)',
          center: 'rgba(194,117,241,1)',
        }}
        activeBgColor={{
          top: 'rgba(206,89,75,0.5)',
          right: 'rgba(246,206,95,0.5)',
          bottom: 'rgba(85,158,101,0.5)',
          left: 'rgba(87,140,238,0.5)',
          center: 'rgba(194,117,241,0.7)',
        }}
        disabledBgColor={{
          top: 'rgba(206,89,75,0.5)',
          right: 'rgba(246,206,95,0.5)',
          bottom: 'rgba(85,158,101,0.5)',
          left: 'rgba(87,140,238,0.5)',
          center: 'rgba(194,117,241,0.7)',
        }}
        disabledColor={{
          center: 'red',
        }}
        status={{
          top: true,
          bottom: true,
          left: true,
          right: true,
          center: true,
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('onPress render', () => {
    const wrapper = shallow(
      <CircleHandle
        tip="channel"
        repeat={true}
        repeatIntervalTime={1000}
        onPress={{
          top: console.log('onPress top'),
          bottom: console.log('onPress bottom'),
          left: console.log('onPress left'),
          right: console.log('onPress right'),
          center: console.log('onPress center'),
        }}
        onLongPress={{
          top: console.log('onLongPress top'),
          bottom: console.log('onLongPress bottom'),
          left: console.log('onLongPress left'),
          right: console.log('onLongPress right'),
          center: console.log('onLongPress center'),
        }}
        status={{
          top: true,
          bottom: false,
          left: true,
          right: true,
          center: false,
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('componentWillReceiveProps0', () => {
    const { wrapper } = customizeLoading({
      loading: { top: false },
    });
    wrapper.setProps({ loading: { top: true } });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
  it('componentWillReceiveProps1', () => {
    const { wrapper } = iconLoading({
      loading: { center: false },
    });
    wrapper.setProps({ loading: { center: true } });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
