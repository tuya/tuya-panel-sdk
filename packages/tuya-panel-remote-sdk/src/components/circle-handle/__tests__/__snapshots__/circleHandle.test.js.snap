// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressKey components componentWillReceiveProps0 1`] = `
<Styled(View)>
  <View
    onLayout={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
  >
    <Image
      source={1}
    />
  </View>
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        null,
      ]
    }
    type={null}
    weight={null}
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          null,
        ]
      }
      type={null}
      weight={null}
    />
    <Styled(View)
      outBgColor="#F6F6F6"
      padding={8}
      radius={200}
    >
      <Styled(View)
        padding={8}
        radius={200}
        style={null}
      >
        <Styled(View)
          padding={8}
          radius={200}
          style={
            Array [
              null,
              Object {
                "overflow": "hidden",
                "transform": Array [
                  Object {
                    "rotate": "45deg",
                  },
                ],
              },
            ]
          }
        >
          <Styled(View)
            padding={8}
            radius={200}
          />
        </Styled(View)>
      </Styled(View)>
      <Styled(View)
        padding={8}
        radius={200}
      />
      <Styled(View)
        centerBgColor="#FFFFFF"
        centerRadius={80}
      />
      <Styled(TouchableHighlight)
        centerBgColor="#FFFFFF"
        centerDisabledBgColor="#FAFAFA"
        centerRadius={80}
        centerStatus={false}
        onLongPress={[Function]}
        onPress={null}
        onPressOut={[Function]}
        radius={200}
        style={null}
        underlayColor="#FAFAFA"
      >
        <View />
      </Styled(TouchableHighlight)>
      <View
        pointerEvents="none"
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          style={
            Array [
              Object {
                "color": "rgba(39,197,153,0.7)",
                "fontSize": 50,
                "fontWeight": "400",
              },
              null,
            ]
          }
          type={null}
          weight={null}
        >
          OK
        </TYText>
      </View>
    </Styled(View)>
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          null,
        ]
      }
      type={null}
      weight={null}
    />
  </View>
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        null,
      ]
    }
    type={null}
    weight={null}
  />
</Styled(View)>
`;

exports[`PressKey components componentWillReceiveProps1 1`] = `
<Styled(View)>
  <View
    onLayout={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
  />
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        null,
      ]
    }
    type={null}
    weight={null}
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          null,
        ]
      }
      type={null}
      weight={null}
    />
    <Styled(View)
      outBgColor="#F6F6F6"
      padding={8}
      radius={200}
    >
      <Styled(View)
        padding={8}
        radius={200}
        style={null}
      >
        <Styled(View)
          padding={8}
          radius={200}
          style={
            Array [
              null,
              Object {
                "overflow": "hidden",
                "transform": Array [
                  Object {
                    "rotate": "45deg",
                  },
                ],
              },
            ]
          }
        >
          <Styled(View)
            padding={8}
            radius={200}
          />
        </Styled(View)>
      </Styled(View)>
      <Styled(View)
        padding={8}
        radius={200}
      />
      <Styled(View)
        centerBgColor="#FFFFFF"
        centerRadius={80}
      />
      <Styled(TouchableHighlight)
        centerBgColor="#FFFFFF"
        centerDisabledBgColor="#FAFAFA"
        centerRadius={80}
        centerStatus={false}
        onLongPress={[Function]}
        onPress={null}
        onPressOut={[Function]}
        radius={200}
        style={null}
        underlayColor="#FAFAFA"
      >
        <AnimatedComponent
          key="center"
          style={
            Object {
              "position": "absolute",
              "transform": Array [
                Object {
                  "rotate": "1deg",
                },
              ],
            }
          }
        >
          <TYText
            align={null}
            color={null}
            size={null}
            style={
              Array [
                Object {
                  "color": "rgba(39,197,153,0.7)",
                  "fontSize": 50,
                  "fontWeight": "400",
                },
                null,
              ]
            }
            type={null}
            weight={null}
          >
            OK
          </TYText>
        </AnimatedComponent>
      </Styled(TouchableHighlight)>
    </Styled(View)>
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          null,
        ]
      }
      type={null}
      weight={null}
    />
  </View>
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        null,
      ]
    }
    type={null}
    weight={null}
  />
</Styled(View)>
`;

exports[`PressKey components content render 1`] = `
<Styled(View)>
  <View
    onLayout={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
  />
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        Object {
          "color": "rgba(194,117,241,1)",
        },
      ]
    }
    type={null}
    weight={null}
  >
    left
  </TYText>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          Object {
            "color": "rgba(194,117,241,1)",
          },
        ]
      }
      type={null}
      weight={null}
    >
      top
    </TYText>
    <Styled(View)
      outBgColor="#F6F6F6"
      padding={8}
      radius={200}
    >
      <Styled(View)
        padding={8}
        radius={200}
        style={null}
      >
        <Styled(View)
          padding={8}
          radius={200}
          style={
            Array [
              null,
              Object {
                "overflow": "hidden",
                "transform": Array [
                  Object {
                    "rotate": "45deg",
                  },
                ],
              },
            ]
          }
        >
          <Styled(View)
            padding={8}
            radius={200}
          />
        </Styled(View)>
      </Styled(View)>
      <Styled(View)
        padding={8}
        radius={200}
      />
      <Styled(View)
        centerBgColor="#FFFFFF"
        centerRadius={80}
      />
      <Styled(TouchableHighlight)
        centerBgColor="#FFFFFF"
        centerDisabledBgColor="#FAFAFA"
        centerRadius={80}
        centerStatus={true}
        onLongPress={[Function]}
        onPress={null}
        onPressOut={[Function]}
        radius={200}
        style={null}
        underlayColor="#FAFAFA"
      >
        <View />
      </Styled(TouchableHighlight)>
      <View
        pointerEvents="none"
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          style={
            Array [
              Object {
                "color": "rgba(39,197,153,1)",
                "fontSize": 50,
                "fontWeight": "400",
              },
              Object {
                "color": "red",
                "fontSize": 32,
              },
            ]
          }
          type={null}
          weight={null}
        >
          OK
        </TYText>
      </View>
    </Styled(View)>
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          Object {
            "color": "rgba(194,117,241,1)",
          },
        ]
      }
      type={null}
      weight={null}
    >
      bottom
    </TYText>
  </View>
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        Object {
          "color": "rgba(194,117,241,1)",
        },
      ]
    }
    type={null}
    weight={null}
  >
    right
  </TYText>
</Styled(View)>
`;

exports[`PressKey components customize render 1`] = `
<Styled(View)>
  <View
    onLayout={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
  />
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        null,
      ]
    }
    type={null}
    weight={null}
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          null,
        ]
      }
      type={null}
      weight={null}
    />
    <Styled(View)
      outBgColor="rgba(246,206,95,0.3)"
      padding={20}
      radius={180}
    >
      <Styled(View)
        padding={20}
        radius={180}
        style={null}
      >
        <Styled(View)
          padding={20}
          radius={180}
          style={
            Array [
              null,
              Object {
                "overflow": "hidden",
                "transform": Array [
                  Object {
                    "rotate": "45deg",
                  },
                ],
              },
            ]
          }
        >
          <Styled(View)
            padding={20}
            radius={180}
          />
        </Styled(View)>
      </Styled(View)>
      <Styled(View)
        padding={20}
        radius={180}
      />
      <Styled(View)
        centerBgColor="rgba(194,117,241,1)"
        centerRadius={60}
      />
      <Styled(TouchableHighlight)
        centerBgColor="rgba(194,117,241,1)"
        centerDisabledBgColor="rgba(194,117,241,0.7)"
        centerRadius={60}
        centerStatus={true}
        onLongPress={[Function]}
        onPress={null}
        onPressOut={[Function]}
        radius={180}
        style={null}
        underlayColor="rgba(194,117,241,0.7)"
      >
        <View />
      </Styled(TouchableHighlight)>
      <View
        pointerEvents="none"
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          style={
            Array [
              Object {
                "color": "rgba(39,197,153,1)",
                "fontSize": 50,
                "fontWeight": "400",
              },
              Object {
                "color": "white",
                "fontSize": 32,
              },
            ]
          }
          type={null}
          weight={null}
        >
          OK
        </TYText>
      </View>
    </Styled(View)>
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          null,
        ]
      }
      type={null}
      weight={null}
    />
  </View>
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        null,
      ]
    }
    type={null}
    weight={null}
  />
</Styled(View)>
`;

exports[`PressKey components onPress render 1`] = `
<Styled(View)>
  <View
    onLayout={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
      }
    }
  />
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        null,
      ]
    }
    type={null}
    weight={null}
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          null,
        ]
      }
      type={null}
      weight={null}
    />
    <Styled(View)
      outBgColor="#F6F6F6"
      padding={8}
      radius={200}
    >
      <Styled(View)
        padding={8}
        radius={200}
        style={null}
      >
        <Styled(View)
          padding={8}
          radius={200}
          style={
            Array [
              null,
              Object {
                "overflow": "hidden",
                "transform": Array [
                  Object {
                    "rotate": "45deg",
                  },
                ],
              },
            ]
          }
        >
          <Styled(View)
            padding={8}
            radius={200}
          />
        </Styled(View)>
      </Styled(View)>
      <Styled(View)
        padding={8}
        radius={200}
      />
      <Styled(View)
        centerBgColor="#FFFFFF"
        centerRadius={80}
      />
      <Styled(TouchableHighlight)
        centerBgColor="#FFFFFF"
        centerDisabledBgColor="#FAFAFA"
        centerRadius={80}
        centerStatus={false}
        onLongPress={[Function]}
        onPress={null}
        onPressOut={[Function]}
        radius={200}
        style={null}
        underlayColor="#FAFAFA"
      >
        <View />
      </Styled(TouchableHighlight)>
      <View
        pointerEvents="none"
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <TYText
          align={null}
          color={null}
          size={null}
          style={
            Array [
              Object {
                "color": "rgba(39,197,153,0.7)",
                "fontSize": 50,
                "fontWeight": "400",
              },
              null,
            ]
          }
          type={null}
          weight={null}
        >
          OK
        </TYText>
      </View>
    </Styled(View)>
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "rgb(104,104,104)",
            "fontSize": 30,
          },
          null,
        ]
      }
      type={null}
      weight={null}
    />
  </View>
  <TYText
    align={null}
    color={null}
    size={null}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgb(104,104,104)",
          "fontSize": 30,
        },
        null,
      ]
    }
    type={null}
    weight={null}
  />
</Styled(View)>
`;
